crates/oxc_linter/src/rules/unicorn/no_null.rs

```diff
@@ -199,7 +199,17 @@ impl Rule for NoNull {
             }
             (AstKind::ReturnStatement(_), _) => {
                 ctx.diagnostic_with_fix(no_null_diagnostic(null_literal.span), |fixer| {
-                    fixer.delete(null_literal)
+                    let mut null_span = null_literal.span;
+                    // Find the last parent that is a TSAsExpression (`null as any`) or TSNonNullExpression (`null!`)
+                    for parent in ctx.nodes().iter_parents(node.id()).skip(1) {
```

Comment:
What happens if there's a null in a call expression? `foo(null) as any[]`

Reply:
Or `foo(null!)`

Reply:
Only `null` in ReturnStatement will be deleted. So:

`foo(null) as any[]` -> `foo(undefined) as any[]`
`foo(null)` -> `foo(undefined)`